src/lib/ui/SupportedDevices/SupportedDevicesFilterArea.tsx

```diff
@@ -0,0 +1,163 @@
+import { capitalize } from '@mui/material'
+import type { Dispatch, SetStateAction } from 'react'
+
+import { CloseIcon } from 'lib/icons/Close.js'
+import { SearchIcon } from 'lib/icons/Search.js'
+import { Button } from 'lib/ui/Button.js'
+import Menu from 'lib/ui/Menu/Menu.js'
+import { FilterCategoryMenu } from 'lib/ui/SupportedDevices/FilterCategoryMenu.js'
+import type { DeviceModel, Filters } from 'lib/ui/SupportedDevices/types.js'
+import { TextField } from 'lib/ui/TextField/TextField.js'
+
+interface SupportedDevicesFilterAreaProps {
+  deviceModels: DeviceModel[]
+  filterStr: string
+  setFilterStr: (filterStr: string) => void
```

Comment:
Prefer to avoid [Hungarian notation](https://en.wikipedia.org/wiki/Hungarian_notation).
```suggestion
  filter: string
  setFilter: (filter: string) => void
```

Or

```
  filterValue: string
  setFilterValue: (filterValue: string) => void
```